compiler/backend/src/org/jetbrains/kotlin/codegen/state/KotlinTypeMapper.kt

```diff
@@ -1199,6 +1199,29 @@ class KotlinTypeMapper @JvmOverloads constructor(
             ?: computeInternalName(classDescriptor, typeMappingConfiguration, isIrBackend)
     }
 
+    private fun getDefaultDescriptor(
+        method: Method,
+        dispatchReceiverDescriptor: String?,
+        callableDescriptor: CallableDescriptor
+    ): String {
+        val descriptor = method.descriptor
+        val extraArgsShift = when {
+            isIrBackend && isConstructor(method) && isEnumClass(callableDescriptor.containingDeclaration) -> 2
+            isIrBackend && isConstructor(method) && (callableDescriptor.containingDeclaration as? ClassDescriptor)?.isInner == true -> 1
+            else -> 0
+        }
+        val maskArgumentsCount = (callableDescriptor.valueParameters.size - extraArgsShift + Integer.SIZE - 1) / Integer.SIZE
```

Comment:
The real change is around `extraArgsShfit`; I have to move this function out of companion in order to use `isIrBackend`.

Reply:
> The real change is around `extraArgsShift`; I have to move this function out of companion in order to use `isIrBackend`.

Thanks for elaborating on this, I was wondering why change location.